crates/api/src/agent.rs

```diff
@@ -0,0 +1,362 @@
+//! Types dealing with agent metadata.
+//!
+//! [AgentInfo] and the wrapping [AgentInfoSigned] define a pattern for
+//! cryptographically verifiable declarations of network reachability.
+//!
+//! To facilitate ease of debugging (See our unit tests in this module!),
+//! The canonical encoding for this info is JSON.
+//!
+//! #### Json Schemas
+//!
+//! ```json
+//! {
+//!   "title": "AgentInfoSigned",
+//!   "type": "object",
+//!   "properties": {
+//!     "agentInfo": { "type": "string", "required": true, "description": "json AgentInfo" },
```

Comment:
technically while it's json, the encoding is specified at a different level, i.e. in the transfer encoding. We cannot specify utf8 here at the schema level, because it might not be.

The fact that we need to make sure to get that string as utf8 encoded when generating or verifying the signature is a separate implementation detail that perhaps should be specified elsewhere in this spec : )